spotify.go

```diff
@@ -124,6 +171,78 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+
+	autoRetry     bool
+	retryDuration time.Duration
+}
+
+func (c *Client) SetAutoRetry(flag bool) {
```

Comment:
Make `autoRetry` exported and remove the setter.  Users can set `client.AutoRetry = true` directly if they want to opt in to this behavior.